prov/verbs/src/ep_rdm/verbs_cq_ep_rdm.c

```diff
@@ -135,8 +133,11 @@ ssize_t fi_ibv_rdm_cq_sreadfrom(struct fid_cq *cq, void *buf, size_t count,
 						    threshold - counter,
 						    src_addr);
 		counter += (ret > 0) ? ret : 0;
-	} while ((ret >= 0) && (counter < threshold ||
-		(timeout >= 0 && fi_gettime_ms() < time_limit)));
```

Comment:
Yes. It should be &&. If timeout occurs, we need to stop reading the CQ even if counter hasn't reached the threshold.
The updated code looks correct.